docs/architecture-multi-standby.rst

```diff
@@ -0,0 +1,329 @@
+.. _multi_node_architecture:
+
+Multi-node Architectures
+========================
+
+Pg_auto_failover allows you to have more than one standby node, and offers
+advanced control over your production architecture characteristics.
+
+Architectures with two standby nodes
+------------------------------------
+
+When adding your second standby node with default settings, you get the
+following architecture:
+
+.. figure:: ./tikz/arch-multi-standby.svg
+   :alt: pg_auto_failover architecture with two standby nodes
+
+   pg_auto_failover architecture with two standby nodes
+
+In this case, three nodes get set up with the same characteristics, achieving
+HA for both the Postgres service and the production dataset. An important
+setting for this architecture is ``number_sync_standbys``.
+
+The replication setting ``number_sync_standbys`` sets how many standby nodes
+the primary should wait for when committing a transaction. In order to have
+a good availability in your system, pg_auto_failover requires
+``number_sync_standbys + 1`` standby nodes participating in the replication
+quorum: this allows any standby node to fail without impact on the system's
+ability to respect the replication quorum.
+
+When only two nodes are registered in a group on the monitor we have a
+primary and a single secondary node. Then ``number_sync_standbys`` can only
+be set to zero. When adding a second standby node to a pg_auto_failover
+group, then the monitor automatically increments ``number_sync_standbys`` to
+one, as we see in the diagram above.
+
+When ``number_sync_standbys`` is set to zero then pg_auto_failover
+implements the *Business Continuity* setup as seen in
+:ref:`architecture_basics`: synchronous replication is then used as a way to
+guarantee that failover can be implemented without data loss.
+
+In more details:
+
+ 1. With ``number_sync_standbys`` set to one, this architecture always
+    maintains two copies of the dataset: one on the current primary node
+    (node A in the previous diagram), and one on the standby that
+    acknowledges the transaction first (either node B or node C in the
+    diagram).
+
+    When one of the standby nodes is unavailable, the second copy of the
+    dataset can still be maintained thanks to the remaining standby.
+
+    When both the standby nodes are unavailable, then it's no longer
+    possible to guarantee the replication quorum, and thus writes on the
+    primary are blocked. The Postgres primary node waits until at least one
+    standby node acknowledges the transactions locally committed, thus
+    degrading your Postgres service to read-only.
+
+ 0. It is possible to manually set ``number_sync_standbys`` to zero when
```

Comment:
I'm using the number of number_sync_standbys as the bullet point numbering system here. I found it a clever thing to do. That said, if that confuses you, there is a good chance that it will confuse every other reader too...